src/general/MusicCategory.cs

```diff
@@ -122,6 +124,12 @@ public void Check()
             track.Check();
     }
 
+    public Track[] GetTracksForContexts(MusicContext[] contexts)
+    {
+        return Tracks.Where(t => t.ExclusiveToContexts == null ||
+            t.ExclusiveToContexts.Any(contexts.Contains)).ToArray();
```

Comment:
This gets called quite a lot, which leads to a bunch of memory allocations. This should be refactored to make it so that tracks can be retrieved without memory allocations, so either return an IEnumerable here or/and add a variant that takes in a list as parameter where the tracks are added.

Also related to the context filtering I think it should be the case that we should have more control on the tracks:
- I think exclusive to context tracks, like you implemented, is good, only play them when a matching context is defined
- But I think non-limited tracks should play even if a context is defined, instead I think there should be a track property "DisallowInContexts" that stops a track playing when some context is defined.
- And there should also be a track property for "PlayOnlyWithoutContext" that only allows playing that track if no context is defined (you seemed to make this the default, but I'm not convinced this should be the default).